### ISSUE TITLE
Document API Routes in POSTMAN and Generate API Documentation

### ISSUE THREAD

**Issue by prasenjeet-symon** [prasenjeet-symon/intellectia#212]
**Title**: Document API Routes in POSTMAN and Generate API Documentation

**Description**:
This GitHub issue focuses on documenting all the existing API routes in POSTMAN to improve the accessibility and understanding of our APIs. It involves creating API documentation with clear descriptions, request examples, and response structures for each route. Additionally, we'll generate API documentation and add a link to the README file for easy reference.

**Tasks**:
1. **API Route Documentation**:
   - [ ] Create POSTMAN documentation for each existing API route.
   - [ ] Include detailed descriptions of each route, specifying the purpose and functionality.
   - [ ] Add request examples to demonstrate how to interact with each API, including required parameters and headers.
   - [ ] Document the expected response structure for each route, indicating the possible status codes and response payloads.

2. **API Documentation Generation**:
   - [ ] Generate API documentation in POSTMAN using the documented API routes.
   - [ ] Ensure that the POSTMAN documentation reflects the latest state of our APIs.

3. **README Update**:
   - [ ] Add a link to the generated API documentation in the README file to make it easily accessible for developers.

**Acceptance Criteria**:
- Each existing API route is documented in POSTMAN with clear descriptions, request examples, and response structures.
- The POSTMAN documentation accurately reflects the latest state of our APIs.
- A link to the generated API documentation is added to the README file for easy reference.

**Additional Information**:
- Documenting our APIs enhances their accessibility and usability.
- Keeping the documentation up-to-date is essential to ensure developers have accurate information.
- The README file serves as a central hub for accessing API documentation.

**Comment by Shuhua-L** [comment]
Hi, I would like to work on this issue, can you assign it to me?

**Comment by prasenjeet-symon** [comment]
@Shuhua-L Yes, assigned to you

**Comment by kenudeh** [comment]
Hi, what is the state of this project now? Still open for contribution?

**Comment by prasenjeet-symon** [comment]
> Hi, what is the state of this project now? Still open for contribution?

Yes it is open . We need clear and concise API documentation.

**Comment by kenudeh** [comment]
Alright, can you please direct me to where the endpoints and other relevant details are defined?

**Comment by prasenjeet-symon** [comment]
> Alright, can you please direct me to where the endpoints and other relevant details are defined?

Check the server folder, there are API routes ( in a specific folder ). And use link on the readme to setup POSTMAN

**Comment by kenudeh** [comment]
Ok, I'll go through everything and get back to you.

**Comment by prasenjeet-symon** [comment]
> Hi Prasenjeet! I can work on this documentation. So far, I have cloned the POSTMAN collection, however, I am having issues cloning the repository to my local machine. I'm getting the following error:
> 
> "[git@github.com](mailto:git@github.com): Permission denied (publickey). fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights and the repository exists."
> 
> I'm still trying to see if I can resolve it myself, but I'd welcome your assistance.

Hi. I think this is the basis of cloning the public repository. Can you watch some YouTube vids that will make it super clear. Thanks

**Comment by kenudeh** [comment]
I've resolved the issue and started the app, but I'm unable to create an account. The "sign up with Goggle" link isn't working either.

**Comment by prasenjeet-symon** [comment]
> I've resolved the issue and started the app, but I'm unable to create an account. The "sign up with Goggle" link isn't working either.

This issue is related to server side , and need no client interaction . Setup `POSTMAN` and start the server that's all. We are working on the client side , but we want to finish server-side task first .

**Comment by kenudeh** [comment]
Just ran into another issue. The server isn't running on localhost:3001 and I believe it has to do with the following error I got:

[nodemon] app crashed - waiting for file changes before starting...

I'm sure it's why I'm not getting any response on POSTMAN, as contained in the README guide.

**Comment by prasenjeet-symon** [comment]
> Just ran into another issue. The server isn't running on localhost:3001 and I believe it has to do with the following error I got:
> 
> [nodemon] app crashed - waiting for file changes before starting...
> 
> I'm sure it's why I'm not getting any response on POSTMAN, as contained in the README guide.

Okay i will take a look . Weather server is running or not .

**Comment by Believet2** [comment]
Hi i will like to be part of this, can you pls assign me?

**Comment by Believet2** [comment]
For the sake of me begin a first timer can i work with someone that can help me? cuz im looking to gain experience and this is my time contributing to an open source project pls.

**Comment by prasenjeet-symon** [comment]
> For the sake of me begin a first timer can i work with someone that can help me? cuz im looking to gain experience and this is my time contributing to an open source project pls.

Do you understand the issue ??  if not plzz ask me any questions you may have . I will help you understand the issue.

**Comment by Believet2** [comment]
> > For the sake of me begin a first timer can i work with someone that can help me? cuz im looking to gain experience and this is my time contributing to an open source project pls.
> 
> Do you understand the issue ?? if not plzz ask me any questions you may have . I will help you understand the issue.

i will be very glad if i can attach with someone to work on this, plzz this is my first time. plzzzzz

**Comment by Believet2** [comment]
> > For the sake of me begin a first timer can i work with someone that can help me? cuz im looking to gain experience and this is my time contributing to an open source project pls.
> 
> Do you understand the issue ?? if not plzz ask me any questions you may have . I will help you understand the issue.

i don't understand the issue very clearly, i need a deep explanation

**Comment by prasenjeet-symon** [comment]
Well there are multiple api routes in `server` folder . Your task to is to add that to postman and also document it to the postman API documentation.

**Comment by Believet2** [comment]
what about cloning the repo to local machine, does it still solve and issue?

**Comment by prasenjeet-symon** [comment]
> what about cloning the repo to local machine, does it still solve and issue?

Have you ever worked with API before?

**Comment by Believet2** [comment]
> > what about cloning the repo to local machine, does it still solve and issue?
> 
> Have you ever worked with API before?

no, that is why i am seeking for someone to work with on this, i just finish learning technical writing and API documentation

**Comment by Believet2** [comment]
> Just ran into another issue. The server isn't running on localhost:3001 and I believe it has to do with the following error I got:
> 
> [nodemon] app crashed - waiting for file changes before starting...
> 
> I'm sure it's why I'm not getting any response on POSTMAN, as contained in the README guide.

Hello @kenudeh can i pair with you to work on this project, i am a newbies and will like someone to work with in other for me to gain experience

**Comment by Believet2** [comment]
So nobody is willing to help me?

**Comment by Believet2** [comment]
Seems the link to the APIs are not working

**Comment by prasenjeet-symon** [comment]
> So nobody is willing to help me?

Sorry about that, Make sure to have a clear and good understanding of API/POSTMAN . You can watch and read good videos about these topics. Thanks

**Comment by prasenjeet-symon** [comment]
> Seems the link to the APIs are not working

We will take a look since I am currently busy with the Sculify Work.

**Comment by kenudeh** [comment]
Hi Prasenjeet-symon, were you able to confirm if the server is running or not? It didn't start on my machine when I tried. That meant I couldn't get any response to plug into POSTMAN.

**Comment by kenudeh** [comment]
@Believet2 just read the instructions contained on the Readme, it's self-explanatory. If I'm able to solve the technical issue I faced, I wouldn't mind carrying you along anyway.

**Comment by prasenjeet-symon** [comment]
> Hi Prasenjeet-symon, were you able to confirm if the server is running or not? It didn't start on my machine when I tried. That meant I couldn't get any response to plug into POSTMAN.

I will check and let you know

**Comment by kenudeh** [comment]
Alright.

**Comment by Believet2** [comment]
> @Believet2 just read the instructions contained on the Readme, it's self-explanatory. If I'm able to solve the technical issue I faced, I wouldn't mind carrying you along anyway.

Alright @prasenjeet-symon

**Comment by Believet2** [comment]
@prasenjeet-symon has the server been resloved?

**Comment by prasenjeet-symon** [comment]
> @prasenjeet-symon has the server been resloved?

Hey @Believet2 can you plzz share the screenshot of error , so that i can take a look.

**Comment by Believet2** [comment]
@prasenjeet-symon 
![Uploading Capture.PNG…]()

**Comment by prasenjeet-symon** [comment]
@Believet2  Image not uploaded successfully. Plzz try again

**Comment by Believet2** [comment]
![Capture](https://github.com/prasenjeet-symon/intellectia/assets/145823631/d55d36d6-8298-4f23-a579-b52f0a06817e)

**Comment by Believet2** [comment]
> ![Capture](https://private-user-images.githubusercontent.com/145823631/295230931-d55d36d6-8298-4f23-a579-b52f0a06817e.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MDk3NjYsIm5iZiI6MTcwNDgwOTQ2NiwicGF0aCI6Ii8xNDU4MjM2MzEvMjk1MjMwOTMxLWQ1NWQzNmQ2LTgyOTgtNGYyMy1hNTc5LWI1MmYwYTA2ODE3ZS5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwOVQxNDExMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZWI0YWU2MzU0MTg1ZDA1YTYwZDcwZmJmZDAyMDZkYTBkZTg3NGY2YWI0MzJmMTFhOWEyZmZkOGI1NGJlMGE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tbKL8CIb1vuCLJxqWbn_tXFCDxXlAkIRtef67g7Qmr8)

@prasenjeet-symon

**Comment by prasenjeet-symon** [comment]
I have tested server is running fine. You need to learn how to setup the postman there is doc on the readme file.  @Believet2

**Comment by Believet2** [comment]
Do I have to install docker on my system? Cuz I have forked the repository in postman and it still shows me same error message

**Comment by satokenta940** [comment]
It's great, but I recommend using [Apidog ](https://www.apidog.com/?utm_source=&utm_medium=blogger&utm_campaign=test1)to record and generate API documentation. I'm currently doing this, and it has been very effective. 
﻿ 
﻿
